src/ScottPlot5/ScottPlot/Plottables/Heatmap.cs

```diff
@@ -0,0 +1,86 @@
+﻿using ScottPlot.Axis;
+using SkiaSharp;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using ScottPlot.Colormaps;
+
+namespace ScottPlot.Plottables
+{
+    public class Heatmap : IPlottable
+    {
+        public bool IsVisible { get; set; } = true;
+        public IAxes Axes { get; set; } = Axis.Axes.Default;
+        public IColormap Colormap { get; set; } = new Viridis();
+
+        private double[,] _intensities;
+        public double[,] Intensities
+        {
+            get => _intensities;
+            set
+            {
+                _intensities = value;
+                Update();
+            }
+        }
+
+        private SKBitmap? bitmap = null;
+
+        public Heatmap(double[,] intensities)
+        {
+            Intensities = intensities;
+        }
+
+        ~Heatmap()
+        {
+            bitmap?.Dispose();
+        }
+
+        public void Update()
+        {
+            bitmap?.Dispose();
+            SKImageInfo imageInfo = new(Intensities.GetLength(0), Intensities.GetLength(1));
+
+            bitmap = new(imageInfo);
+            var intensitiesFlat = Intensities.Cast<double>();
```

Comment:
The cast is likely unnecessary, `double[,]` implements `IEnumerable<double>`, not `IEnumerable<double[]>` or similar.

Reply:
Sike, it implements `IEnumerable` but not `IEnumerable<double>` even though each element it returns is of type `double`